cylc/flow/config_schema.py

```diff
@@ -0,0 +1,609 @@
+                'speedup factor': [VDR.V_FLOAT],
+                'time limit buffer': [VDR.V_INTERVAL, DurationFloat(30)],
+                'fail cycle points': [VDR.V_STRING_LIST],
+                'fail try 1 only': [VDR.V_BOOLEAN, True],
+                'disable task event handlers': [VDR.V_BOOLEAN, True],
+            },
+            'environment filter': {
+                'include': [VDR.V_STRING_LIST],
+                'exclude': [VDR.V_STRING_LIST],
+            },
+            'job': {
+                'execution retry delays': [VDR.V_INTERVAL_LIST, None],
+                'execution time limit': [VDR.V_INTERVAL],
+            },
+            'platform': [VDR.V_STRING],
+            'events': {
+                'execution timeout': [VDR.V_INTERVAL],
+                'handlers': [VDR.V_STRING_LIST, None],
+                'handler events': [VDR.V_STRING_LIST, None],
+                'handler retry delays': [VDR.V_INTERVAL_LIST, None],
+                'mail events': [VDR.V_STRING_LIST, None],
+                'mail from': [VDR.V_STRING],
+                'mail retry delays': [VDR.V_INTERVAL_LIST, None],
+                'mail smtp': [VDR.V_STRING],
+                'mail to': [VDR.V_STRING],
+                'submission timeout': [VDR.V_INTERVAL],
+
+                'expired handler': [VDR.V_STRING_LIST, None],
+                'late offset': [VDR.V_INTERVAL, None],
+                'late handler': [VDR.V_STRING_LIST, None],
+                'submitted handler': [VDR.V_STRING_LIST, None],
+                'started handler': [VDR.V_STRING_LIST, None],
+                'succeeded handler': [VDR.V_STRING_LIST, None],
+                'failed handler': [VDR.V_STRING_LIST, None],
+                'submission failed handler': [VDR.V_STRING_LIST, None],
+                'warning handler': [VDR.V_STRING_LIST, None],
+                'critical handler': [VDR.V_STRING_LIST, None],
+                'retry handler': [VDR.V_STRING_LIST, None],
+                'submission retry handler': [VDR.V_STRING_LIST, None],
+                'execution timeout handler': [VDR.V_STRING_LIST, None],
+                'submission timeout handler': [VDR.V_STRING_LIST, None],
+                'custom handler': [VDR.V_STRING_LIST, None],
+            },
+            'suite state polling': {
+                'user': [VDR.V_STRING],
+                'host': [VDR.V_STRING],
+                'interval': [VDR.V_INTERVAL],
+                'max-polls': [VDR.V_INTEGER],
+                'message': [VDR.V_STRING],
+                'run-dir': [VDR.V_STRING],
+                'verbose mode': [VDR.V_BOOLEAN],
+            },
+            'environment': {
+                '__MANY__': [VDR.V_STRING],
+            },
+            'directives': {
+                '__MANY__': [VDR.V_STRING],
+            },
+            'outputs': {
+                '__MANY__': [VDR.V_STRING],
+            },
+            'parameter environment templates': {
```

Comment:
`task parameter ...`  (see comment above on parameters not been a general templating mechanism ... although here it's in a task config section so maybe clear enough as-is).

Reply:
I'll leave this one open @wxtim because you have already implemented the corresponding items: `task parameters` and `task parameter templates`.

Reply:
Ping @wxtim - this one still active.